### PR TITLE
Some minor simplifications & optimizations

### PR DESCRIPTION
**Pull request by MaxGraey**

### CODE REVIEW

**Comment by ColinEberhardt** [comment]
Thanks @MaxGraey - really appreciate this. I'll take a look over these shortly, make sure i full digest them

File: assembly/parser/node.ts [hunk 1]
```diff
@@ -1,12 +1,22 @@
-import { replaceAtIndex, toArray } from "../util";
+import { replaceAtIndex } from "../util";
 
 const emptyNodeArray = new Array<Node>();
 
+export const enum NodeType {
```
Comment (ColinEberhardt) [inline assembly/parser/node.ts:5]: @MaxGraey thanks - been meaning to do this for a while 👍

File: assembly/util.ts [hunk 2]
```diff
@@ -1,20 +1,17 @@
...
 export function first<T>(arr: T[]): T {
-  return arr[0];
+  return unchecked(arr[0]);
 }
 
 export function toArray<T>(item: T): T[] {
-  const arr = new Array<T>();
-  arr.push(item);
-  return arr;
+  return [item];
```
Comment (ColinEberhardt) [inline assembly/util.ts:10]: For some reason I didn't think Array literals were supported in AS! that does render this helper pretty much redundant

**Review by ColinEberhardt — APPROVED** [review]
Thanks - loads of good stuff in here and saves me a lot of refactoring and tidyup :-)

I noticed a few simple changes with performance in mind. I think I'll make writing a benchmark my next step so that I can more effectively evaluate these types of change.

File: assembly/nfa/characters.ts [hunk 3]
```diff
@@ -1,32 +1,34 @@
...
+  switch (code) {
+    case 0x1680: // <LS> (1)
+    case 0x2028: // <LS> (2)
+    case 0x2029: // <PS>
+    case 0x202F: // <NNS>
+    case 0x205F: // <MMSP>
+    case 0x3000: // <IS>
+    case 0xFEFF: return true; // <ZWNBSP>
+  }
+  return false;
```
Comment (MaxGraey) [inline assembly/nfa/characters.ts:33]: note all this code is not properly transpilable to JS by `tsc` due to based on unsigned integer operations (code has `u32`). If you want more portable code you probably to follow two different approaches:
1) Add JS-style casts after every relational ops. like: `((code - 0x2000) >>> 0) <= ((0x200A - 0x2000) >>> 0)`. 
2) Or you could use AS macro:
```ts
if (ASC_TARGET == 1) {
  // AS code with unsigned ops
} else {
  // JS compatible code which use `>>> 0` or different implementation
}
```

Comment (ColinEberhardt) [reply assembly/nfa/characters.ts:33]: Thanks for pointing this out. I do make extensive use of the ability to run this code as TS=>JS as well as AS=>wasm - I find the debugging experience much easier with TS/JS!

**Pull request merged**